src/mlpack/core/optimizers/Nadam/nadam_update.hpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file nadam_update.hpp
+ * @author Sourabh Varshney
+ *
+ * Nadam optimizer. Nadam is an optimizer that combines the effect of Adam and 
+ * NAG to the gradient descent to improve its Performance.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_NADAM_NADAM_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_NADAM_NADAM_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Nadam is an optimizer that combines the Adam and NAG.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{
+ *   author  = {Sebastian Ruder},
+ *   title   = {An overview of gradient descent optimization algorithms},
+ *   journal = {CoRR},
+ *   year    = {2016},
+ *   url     = {https://arxiv.org/abs/1609.04747v2}
+ * }
+ * @endcode
+ */
+class NadamUpdate
+{
+ public:
+  /**
+   * Construct the Nadam update policy with the given parameters.
+   *
+   * @param epsilon The epsilon value used to initialise the squared gradient
+   *        parameter.
+   * @param beta1 The smoothing parameter.
+   */
+  NadamUpdate(const double epsilon = 1e-8, const double beta1 = 0.9)
+  :epsilon(epsilon), beta1(beta1), iteration(0)
+  {
+    // Nothing to do.
+  }
+
+  /**
+   * The Initialize method is called by SGD Optimizer method before the start 
+   * of the iteration update process.
+   *
+   * @param rows Number of rows in the gradient matrix.
+   * @param cols Number of columns in the gradient matrix.
+   */
+  void Initialize(const size_t rows, const size_t cols)
+  {
+    m = arma::zeros<arma::mat>(rows, cols);
+    v = arma::zeros<arma::mat>(rows, cols);
+  }
+
+  /**
+   * Update step for Nadam.
+   *
+   * @param iterate Parameters that minimize the function.
+   * @param stepSize Step size to be used for the given iteration.
+   * @param gradient The gradient matrix.
+   */
+  void Update(arma::mat& iterate, const double stepSize,
+  const arma::mat& gradient)
+  {
+    // Increment the iteration counter variable.
+    ++iteration;
+
+    // And update the iterate.
+    m *= beta1;
+    m += (1 - beta1) * gradient;
+    // biasCorrection=1-beta1^iteration
+    const double biasCorrection = 1.0 - std::pow(beta1, iteration);
+    /*
+	iterate=iterate-((stepsize/(sqrt(v)+epsilon))*(m/biasCorrection))
+    */
+    iterate -= ((stepSize * m)/(biasCorrection1 * (arma::sqrt(v) + epsilon)));
```

Comment:
Do you mind to follow the update rule as described in the paper: https://openreview.net/pdf?id=OM0jvwB8jIp57ZJjtNEZ, which also uses the decay parameter, and also updates the second momentum vector, in your code ``v``is always zero.

Reply:
I would not mind it. I would follow this and update my code by tomorrow.

Reply:
The paper which I referred did not took account for that. I would go further deep into the details and would change it accordingly.